### PR TITLE
Add plugin to enable/disable the generation of targetSchema entry for…

### PR DESCRIPTION
**Pull request by skynetsonar**
… a link

### CODE REVIEW

**Review by florianschrack — COMMENTED** [review]
Looks good to me apart from the minor javadoc issue

File: src/test/java/com/mercateo/common/rest/schemagen/link/LinkCreatorTest.java [hunk 1]
```diff
@@ -190,7 +200,7 @@ protected Something getReturnType(ImplementedBeanParamType param) {
     }
 
     @Test
-    public void testTargetSchemaPresentOnMatchingMediaTypeAtMethodLevel()
+    public void testTargetSchemaPresentOnMatchingMediaTypeAtMethodLevel__targetSchemaEnabled()
```
Comment (skynetsonar) [inline src/test/java/com/mercateo/common/rest/schemagen/link/LinkCreatorTest.java:203]: Underscore in method name does not match the naming scheme of the other test methods

File: src/main/java/com/mercateo/common/rest/schemagen/link/LinkCreator.java [hunk 2]
```diff
@@ -236,11 +236,13 @@ private void addSchemaIfNeeded(Builder builder, Scope method,
         Optional<String> optionalInputSchema = jsonSchemaGenerator.createInputSchema(method,
                 linkFactoryContext.getFieldCheckerForSchema());
         optionalInputSchema.ifPresent(s -> builder.param(SCHEMA_PARAM_KEY, s));
-        Optional<String> mt = detectMediaType(method.getInvokedMethod());
-        if (mt.isPresent() && MediaType.APPLICATION_JSON.equals(mt.get())) {
-            Optional<String> optionalOutputSchema = jsonSchemaGenerator.createOutputSchema(method,
-                    linkFactoryContext.getFieldCheckerForSchema());
-            optionalOutputSchema.ifPresent(s -> builder.param(TARGET_SCHEMA_PARAM_KEY, s));
+        if (linkFactoryContext.getTargetSchemaEnablerForLink().test(method)) {
```
Comment (JoergAdler) [inline src/main/java/com/mercateo/common/rest/schemagen/link/LinkCreator.java:239]: just one question here: why check this in an outer if here, while 

jsonSchemaGenerator.createOutputSchema(method,
                        linkFactoryContext.getFieldCheckerForSchema());

already has an optional return type?

File: src/test/java/com/mercateo/common/rest/schemagen/link/LinkCreatorTest.java (hunk 1 above)
Comment (skynetsonar) [reply src/test/java/com/mercateo/common/rest/schemagen/link/LinkCreatorTest.java]: done

File: src/main/java/com/mercateo/common/rest/schemagen/link/LinkMetaFactory.java [hunk 3]
```diff
@@ -73,26 +74,31 @@ public static LinkMetaFactory create(JsonSchemaGenerator jsonSchemaGenerator, Li
...
 		requireNonNull(methodCheckerForLink);
 		requireNonNull(fieldCheckerForSchema);
-		return new LinkMetaFactory(jsonSchemaGenerator, new LinkFactoryContextDefault(baseUri, methodCheckerForLink, fieldCheckerForSchema));
+		return new LinkMetaFactory(jsonSchemaGenerator, new LinkFactoryContextDefault(baseUri, methodCheckerForLink, fieldCheckerForSchema,
+				targetSchemaEnablerForLink));
 	}
 
 	/**
 	 * @deprecated should not be used any more, because the use of a global link factory context is deprecated
+	 * @param targetSchemaEnablerForLink
```
[the remark this answers is not in the document]
Comment (skynetsonar) [reply src/main/java/com/mercateo/common/rest/schemagen/link/LinkMetaFactory.java]: done (should have actually been automatically updated )

File: src/main/java/com/mercateo/common/rest/schemagen/link/LinkCreator.java (hunk 2 above)
Comment (skynetsonar) [reply src/main/java/com/mercateo/common/rest/schemagen/link/LinkCreator.java]: you are right. done

**Pull request merged**